File: backend/src/main/java/com/dream/services/TaskService.java
package com.dream.services;

import org.springframework.data.domain.Pageable;

import org.springframework.data.domain.Page;

import com.dream.models.Task;

public interface TaskService {
	
	Page<Task> getTasks(Pageable pageable);
	
	Task getTaskById(long id);
	
	Task createTask(Task Task);
	
	Task updateTask(long id, Task Task);
	
	void deleteTask(long id);
}

File: frontend/src/app/services/administration/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Task } from '../../model/task.model';
import { API_URL } from '../../app.constants';

@Injectable({
  providedIn: 'root'
})
export class TaskService {

  constructor(
    private http:HttpClient
  ) { }

  retrieveAllTasks(page,size,sort) {
    return this.http.get<Task[]>(`${API_URL}/tasks?page=${page}&size=${size}&sort=${sort}`);
  }

  deleteTask(taskId){
    return this.http.delete(`${API_URL}/tasks/${taskId}`);
  }

  retrieveTask(taskId){
    return this.http.get<Task>(`${API_URL}/tasks/${taskId}`);
  }

  updateTask(taskId, task){
    return this.http.put(
          `${API_URL}/tasks/${taskId}`
                , task);
  }

  createTask(task){
    return this.http.post(
              `${API_URL}/tasks`
                , task);
  }
}
